### ISSUE TITLE
serverless dinitctl

### ISSUE THREAD

**Issue by jackghg** [davmac314/dinit#86]
In a installation dinit has never been launched yet, so I cant enable important services like networkmanager, ssh... (I dont remember this problem before, is this a new change?)

I'd like to dinitctl to enable/disable services without dinit server, or at least enable... if I execute the command as root. By default or with a option ? or maybe a workaround ?
Thanks.

**Comment by davmac314** [comment]
I do want this to be possible, but no, it's never worked that way.

The workaround I guess is to create the link yourself. I.e. `dinitctl enable xyz` just looks in the boot service for a `waitsfor.d =` directive, then creates a symbolic link in that directory (with the name `xyz`, with the link referring to the `xyz` service description file, although it doesn't actually matter what it links to).

**Comment by mobin-2008** [comment]
> In a installation dinit has never been launched yet, so I cant enable important services like networkmanager, ssh... (I dont remember this problem before, is this a new change?)

What do you mean by "installation"? Artix installer?

**Comment by jackghg** [comment]
Ok.


> `dinitctl enable xyz` just looks in the boot service for a `waitsfor.d =` directive to find a directory, then creates a symbolic link in that directory (with the name `xyz`, with the link referring to the `xyz` service description file

Is this what u want to do or what it does today ? because for me it doesnt enable the service, gust give an error messagge.

**Comment by jackghg** [comment]
> What do you mean by "installation"? Artix installer?

In my case i was converting arch to artix, but maybe the same problem can happend when chrooting ?

**Comment by mobin-2008** [comment]
@jackghg Dinit can run in System mode with the following command (not as System-Manager):
`dinit -s`
This commands launch Dinit, but Dinit will no longer manage things like reboot, shutdown, then you can use it with the `dinitctl` command.
The only problem is that I don't know, this command mount `dinit socket` at /run/dinitctl or not? So I'm not sure this is the solution. (you can test it)
from dinit man page:
>[-s, --system](https://davmac.org/projects/dinit/man-pages-html/dinit.8.html#s)
>    Run as the system service manager. This is the default if invoked as the root user. This option affects the default service definition directory and control socket path.
> [-m, --system-mgr](https://davmac.org/projects/dinit/man-pages-html/dinit.8.html#m)
>    Run as the system manager (perform operations directly related to machine startup and shutdown). This is the default when running as process ID 1. The main user-visible effect of this option is to invoke the shutdown program when a shutdown is requested (and after all services have stopped), and to provide some basic support for system recovery in case the boot service (or other specified service) cannot be started.


**Note: for access to terminal after this command, you need add `&` to command: `dinit -s &`**

**Comment by jackghg** [comment]
> Dinit can run

ok, tnx, ill test. however its a bit complex workaround... like make the symlink is easyer... (but dirtyer maybe)

**Comment by mobin-2008** [comment]
> > Dinit can run
> 
> ok, tnx, ill test. however its a bit complex workaround... like make the symlink is easyer... (but dirtyer maybe)

It might sound "complicated", but it's really simple:
1- Start the dinit daemon via `dinit -s &` command
2- Work normally with the `dinitctl` command
3- Finally, close the dinit daemon via `dinitctl shutdown` command.
Done! It should be almost like this.

**Comment by mobin-2008** [comment]
Of course, doing some things (such as enabling services) is easier to do manually But it doesn't work to start or stop the service.

**Comment by davmac314** [comment]
I will implement a flag to allow use of `dinitctl enable` when dinit isn't running at some point.

> Is this what u want it to do or what it does today ? because for me it doesnt enable the service, just give an error messagge.

That's what it does when dinit is running (as well as informing dinit of the changing dependency). When dinit is not running it will give an error as you say, but I will look at allowing this at some point (via a command-line switch).

**Comment by jackghg** [comment]
@mobin-2008 it worked... but i still want this feature :)

**Comment by jackffmm** [comment]
I'm jackghg, this is my new account.

**Comment by jackffmm** [comment]
you did it ?

**Comment by davmac314** [comment]
Addressed in 788d65ff99b991284b3f782eabf5c7dc5145474d

**Comment by jackffmm** [comment]
u r amazing thanks!
